plugins/updater/src/updater.rs

```diff
@@ -558,41 +573,90 @@ impl Update {
-            unsafe {
-                ShellExecuteW(
-                    0,
-                    w!("open"),
-                    file.as_ptr(),
-                    parameters.as_ptr(),
-                    std::ptr::null(),
-                    SW_SHOW,
-                )
-            };
+        let file = encode_wide(path.as_os_str());
+        let parameters = encode_wide(installer_args.join(OsStr::new(" ")).as_os_str());
+        unsafe {
+            ShellExecuteW(
+                0,
+                w!("open"),
+                file.as_ptr(),
+                parameters.as_ptr(),
+                std::ptr::null(),
+                SW_SHOW,
+            )
+        };
+        std::process::exit(0);
+    }
 
-            std::process::exit(0);
+    #[cfg(windows)]
+    fn extract_zip(bytes: &Vec<u8>) -> Result<WindowsUpdaterType> {
+        #[cfg(feature = "zip")]
+        {
+            let tmp_dir = tempfile::Builder::new().tempdir()?.into_path();
+            let archive = Cursor::new(bytes);
+            let mut extractor = zip::ZipArchive::new(archive)?;
+            extractor.extract(&tmp_dir)?;
+
+            let paths = std::fs::read_dir(&tmp_dir)?;
+
+            for path in paths {
+                let found_path = path?.path();
+                if found_path.extension() == Some(OsStr::new("exe")) {
+                    return Ok(WindowsUpdaterType::Nsis {
+                        path: found_path,
+                        temp_path: None,
+                    });
+                } else if found_path.extension() == Some(OsStr::new("msi")) {
+                    return Ok(WindowsUpdaterType::Msi {
+                        path: found_path,
+                        temp_path: None,
+                    });
+                }
+            }
+            return Err(crate::Error::BinaryNotFoundInArchive);
         }
+        #[cfg(not(feature = "zip"))]
+        return Err(crate::Error::InvalidUpdaterFormat);
+    }
 
-        Ok(())
+    #[cfg(windows)]
+    fn extract_executable(bytes: &Vec<u8>) -> Result<WindowsUpdaterType> {
+        use std::io::Write;
+        if infer::app::is_exe(&bytes) {
+            let mut temp_file = tempfile::Builder::new().suffix(".exe").tempfile()?;
+            temp_file.write_all(&bytes)?;
+            let temp_path = temp_file.into_temp_path();
+            return Ok(WindowsUpdaterType::Nsis {
+                path: temp_path.to_path_buf(),
+                temp_path: Some(temp_path),
+            });
+        }
+        if infer::archive::is_msi(&bytes) {
+            let mut temp_file = tempfile::Builder::new().suffix(".msi").tempfile()?;
+            temp_file.write_all(&bytes)?;
+            let temp_path = temp_file.into_temp_path();
+            return Ok(WindowsUpdaterType::Msi {
+                path: temp_path.to_path_buf(),
+                temp_path: Some(temp_path),
+            });
+        }
+        return Err(crate::Error::InvalidUpdaterFormat);
```

Comment:
use if else instead of multiple return statements
```suggestion
        if infer::app::is_exe(&bytes) {
            let mut temp_file = tempfile::Builder::new().suffix(".exe").tempfile()?;
            temp_file.write_all(&bytes)?;
            let temp_path = temp_file.into_temp_path();
            Ok(WindowsUpdaterType::Nsis {
                path: temp_path.to_path_buf(),
                temp_path: Some(temp_path),
            })
        } else if infer::archive::is_msi(&bytes) {
            let mut temp_file = tempfile::Builder::new().suffix(".msi").tempfile()?;
            temp_file.write_all(&bytes)?;
            let temp_path = temp_file.into_temp_path();
            Ok(WindowsUpdaterType::Msi {
                path: temp_path.to_path_buf(),
                temp_path: Some(temp_path),
            })
        } else {
            Err(crate::Error::InvalidUpdaterFormat)
        }
```